front_end/html/exercise_shared.html

```diff
@@ -106,131 +157,72 @@
-                        code_output += build_image_output(json_data.image_output);
-                        document.getElementById("code_output").innerHTML = code_output;
-                    {% end %}
-                }
+                function submit() {
+                    user_code = editor.getValue();
 
-                function run_code() {
-                    user_code = editor.getSelectedText();
-                    if (user_code.trim() == "")
-                        user_code = editor.getValue();
                     if (user_code.trim() == "")
                         return false;
 
+                    // get partner_id
+                    let partner_key = document.getElementById("pair_programming").value == "Yes" ? document.getElementById("partner_key").value : null;
+
                     result_div = document.getElementById("result");
                     result_div.innerHTML = "<div class='notification is-warning is-light smaller-font'>Please wait while your solution is being executed...</div>";
                     result_div.scrollIntoView();
-                    document.getElementById("code_output").innerHTML = "";
 
-                    {% if exercise_details["show_expected"] %}
-                        document.getElementById("diff_output").innerHTML = "";
+                    {% if exercise_details["back_end"] != "any_response" %}
+                        document.getElementById("code_output").innerHTML = "";
+
+                        {% if exercise_details["show_expected"] %}
+                            document.getElementById("diff_output").innerHTML = "";
+                        {% end %}
                     {% end %}
 
-                    post_data = { "user_code": user_code };
+                    post_data = { "partner_key": partner_key, "user_code": user_code, "date": new Date().toLocaleString("en-US", {timeZone: "UTC"})};
 
-                    $.post("/run_code/{{ course_basics['id'] }}/{{ assignment_basics['id'] }}/{{ exercise_basics['id'] }}",
+                    $.post("/submit/{{ course_basics['id'] }}/{{ assignment_basics['id'] }}/{{ exercise_basics['id'] }}",
                         post_data,
-                        function(data, status) {
-                            var json_data = JSON.parse(data);
-
-                            var code_output = "<h6>Your output:</h6>\n\n";
-                            code_output += build_text_output(json_data.text_output);
-                            code_output += build_image_output(json_data.image_output);
-                            document.getElementById("code_output").innerHTML = code_output;
-
-                            document.getElementById("result").innerHTML = "";
-                        })
-                    editor.focus();
-                    return false;
-                 }
-
-                 submitButton = document.getElementById("submit_button");
+                       function(data, status) {
```

Comment:
It looks like some extra stuff from another branch got included in this branch? Or is this all related to the "Copy exercise" functionality?